### PR TITLE
use ogdi.pc if available

### PR DESCRIPTION
**Pull request by praiskup**
This is to avoid patching systems where gdal has to depend
transitively on libtirpc (through libogdi).

## What does this PR do?

Fixes build system issues.

## What are related issues/pull requests?

None.

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

## Environment

Provide environment details, if relevant:

* OS: Linux
* Compiler:

### CODE REVIEW

**Comment by praiskup** [comment]
btw.: git show --ignore-space-change

**Comment by praiskup** [comment]
Alson $LIBS updated now.  Plus I added one commit with `configure` and `aclocal.m4` change - but that really only bloats the git size :-(.

**Comment by rouault** [comment]
This causes errors in Travis-CI tests when --with-ogdi is not specified, and ogdi.pc is not found. The expected and current behaviour when no explicit switch is provided is that ogdi should be used if detected, and skipped without error otherwise

**Comment by praiskup** [comment]
@rouault 
> The expected and current behaviour when no explicit switch is provided is that ogdi should be used if detected, and skipped without error otherwise

Sorry, I forgot that PKG_CHECK_MODULES fails by default.  Fixed, thanks!

File: gdal/configure.ac [hunk 1]
```diff
@@ -1968,33 +1968,38 @@ if test "$with_ogdi" = "no" ; then
...
+    if test "$ac_cv_header_ecs_h" = "no" ; then
+      if test -f "/usr/include/ogdi/ecs.h"; then
+          OGDI_INCLUDE="-I/usr/include/ogdi"
+      elif test -f "/usr/local/include/ogdi/ecs.h"; then
+          OGDI_INCLUDE="-I/usr/local/include/ogdi"
+      elif test -f "/usr/include/ecs.h"; then
+          OGDI_INCLUDE="-I/usr/include"
+      elif test -f "/usr/local/include/ecs.h"; then
+          OGDI_INCLUDE="-I/usr/local/include"
+      fi
```
Comment (praiskup) [inline gdal/configure.ac]: Oh nice, there's the "diff settings" above with "Hide whitespace changes"!

**Pull request merged**